git/github/github_test.go

```diff
@@ -195,4 +232,65 @@ var _ = Describe("Client", func() {
 		Expect(err).To(BeNil())
 		Expect(checkRunID).To(BeNil())
 	})
+
+	It("can check if check run updated is needed", func() {
+		var checkRunAdapter = &github.CheckRunAdapter{
+			Name:           "example-name",
+			Owner:          "example-owner",
+			Repository:     "example-repo",
+			SHA:            "abcdef1",
+			ExternalID:     "example-external-id",
+			Conclusion:     "success",
+			Title:          "example-title",
+			Summary:        "example-summary",
+			Text:           "example-text",
+			StartTime:      time.Now(),
+			CompletionTime: time.Now(),
+		}
+
+		allCheckRuns, err := client.GetAllCheckRunsForRef(context.TODO(), "", "", "", 1)
+		Expect(err).To(BeNil())
+		Expect(len(allCheckRuns) > 0).To(BeTrue())
+
+		existingCheckRun := client.GetExistingCheckRun(allCheckRuns, checkRunAdapter)
+		Expect(existingCheckRun).NotTo(BeNil())
+		Expect(client.IsUpdateNeeded(existingCheckRun, checkRunAdapter)).To(BeTrue())
+
+		checkRunAdapter = &github.CheckRunAdapter{
+			Name:           "example-name",
+			Owner:          "example-owner",
+			Repository:     "example-repo",
+			SHA:            "abcdef1",
+			ExternalID:     "example-external-id",
+			Conclusion:     "failure",
+			Title:          "example-title",
+			Summary:        "example-summary",
+			Text:           "example-text-update",
+			StartTime:      time.Now(),
+			CompletionTime: time.Now(),
+		}
+		Expect(client.IsUpdateNeeded(existingCheckRun, checkRunAdapter)).To(BeFalse())
+	})
+
+	It("can check if creating a new commit status is needed", func() {
+		commitStatuses, err := client.GetAllCommitStatusesForRef(context.TODO(), "", "", "")
+		Expect(err).To(BeNil())
+		Expect(len(commitStatuses) > 0).To(BeTrue())
+
+		commitStatusExist, err := client.CommitStatusExists(commitStatuses, commitStatusAdapter)
+		Expect(commitStatusExist).To(BeTrue())
+		Expect(err).To(BeNil())
+
+		commitStatusAdapter = &github.CommitStatusAdapter{
+			Owner:       "example-owner",
+			Repository:  "example-repo",
+			SHA:         "abcdef1",
+			State:       "Fail",
```

Comment:
I see that Conclusion and State equals in previous case, can we do this here also?
Set both Conclusion and State to **Fail?**